### PR TITLE
[AN-5812][AN-5813] the user can see and update the availability status

### PR DESCRIPTION
**Pull request by makingthematrix**
Use this link to download APK:
https://drive.google.com/open?id=1Vw1bgz8ReQqNAogdhwuK_17EyaFQL5Cy

#### APK
[Download build #9978](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9978/artifact/build/artifact/wire-dev-PR1318-9978.apk)
[Download build #9979](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9979/artifact/build/artifact/wire-dev-PR1318-9979.apk)
[Download build #9981](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9981/artifact/build/artifact/wire-dev-PR1318-9981.apk)

### CODE REVIEW

**Comment by someoneEsle** [comment]
Bug 1
You can only set the status if you already have one (that is, changing status is okay but setting a status for the first time won't work). Therefore I cannot set a status on a fresh install --> I can't test anything else :D

**Comment by makingthematrix** [comment]
ok, check the current APK

**Comment by someoneEsle** [comment]
Bug 2
After a fresh install you can only set the status by tapping the top bar in the conversations list, but there is no way to set it from the profile page

**Comment by someoneEsle** [comment]
Bug 3
Status is not synced across devices. I'm using two android devices with the same apk, logged in with the same account, and changing the status in one of the devices does not change it on the other

**Comment by someoneEsle** [comment]
Bug 4
You can't see statuses in the start UI

Bug 5
the icon status in the conversations list overlaps with the subtitles of the conversations

**Comment by someoneEsle** [comment]
Bug 6
Can we make the little arrow on the profile page clickable too? It's screaming to be clicked but it doesn't seem to be responsive :)

**Comment by someoneEsle** [comment]
Bug 7
Log in with two accounts, T and P. T is a team account. Switch to account P. Update the status of T from another device. Now you see the update dot on the icon for T in the profile view, as if there was a new message (but there isn't, you just updated your status).

**Comment by someoneEsle** [comment]
Bug 8
On the profile page, Manage team and the status are too close (especially on small screens). Log in with an admin to see this

**Comment by someoneEsle** [comment]
Bug 9
Personal accounts can set a status in the profile page (this is probably just a UI problem, since their contacts don't see the status update)

**Comment by someoneEsle** [comment]
Bug 10
If I set my status while I'm offline, the status is not sent once I go back online (row 34 of gdoc)

**Comment by someoneEsle** [comment]
Bug 12
There's no tracking anywhere (conversations list or profile page)

**Comment by someoneEsle** [comment]
Note to self: test what happens when logging out, status changes on another device, log back in

**Comment by someoneEsle** [comment]
Bug 13
the status icon and the name of the person are too close in the start UI (contacts list)

**Comment by someoneEsle** [comment]
Bug 14
When I open the profile details of a contact that has set their status, there shouldn't be an arrow to indicate a menu (like if I could change the other person's status)

**Comment by someoneEsle** [comment]
Bug 15
Tracking always says  "method": "settings, list_header" even if I set the status from the profile page

**Comment by eike96** [comment]
- [ ] in the group details view, the status icon is too far away from the name
- [ ] "sat a status" in the profile and your name in the list UI header are too hard to tap. increase tap area

**Comment by someoneEsle** [comment]
Bug 16
The "TIMED MESSAGE" placeholder should have the accent color.

**Comment by someoneEsle** [comment]
Bug 17
Design just asked us to display the statuses in the following order (from top to bottom): Available, Away, Busy, None.

**Comment by someoneEsle** [comment]
Bug 17
The order should be (from top to bottom): None, Available, Busy, Away

File: app/src/main/scala/com/waz/zclient/cursor/CursorView.scala [hunk 1]
```diff
@@ -196,27 +199,40 @@ class CursorView(val context: Context, val attrs: AttributeSet, val defStyleAttr
     hintView.setTranslationX(pos)
   }
 
-
-  val hintText = controller.convIsEphemeral map {
-    case true => R.string.cursor__ephemeral_message
-    case false => R.string.cursor__type_a_message
+  private val transformer = TextTransform.get(context.getResources.getString(R.string.single_image_message__name__font_transform))
```
Comment (JPonte) [inline app/src/main/scala/com/waz/zclient/cursor/CursorView.scala]: Use ContextUtils to get the string.

File: app/src/main/scala/com/waz/zclient/views/AvailabilityView.scala [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  val white = Color.argb(255, 255, 255, 255)
+  val black = Color.argb(255, 0, 0, 0)
+
+  def drawable(availability: Availability, res: Resources, fillColor: Int = white, dpSize: Int = 10): Option[Drawable] = availability match {
+    case Availability.None => None
+    case _ =>
+      val drawable = iconCache.getOrElseUpdate(s"${availability.id}.$dpSize.$fillColor", {
+        val bmp = Bitmap.createBitmap(64, 64, Bitmap.Config.ARGB_8888)
+        viewData(availability).drawIcon(new Canvas(bmp), fillColor)
+        val px = dpSize * (res.getDisplayMetrics.xdpi / DisplayMetrics.DENSITY_DEFAULT).round
```
Comment (JPonte) [inline app/src/main/scala/com/waz/zclient/views/AvailabilityView.scala]: Consider passing Context as an implicit parameter and use ContextUtils for these things.

**Comment by someoneEsle** [comment]
Bug 18
All the names in the conversations list have been moved up -- they are not in the center anymore

**Pull request merged**